### PR TITLE
Add tests for the new cookie banner

### PR DESCRIPTION
**Pull request by walaura**
## What does this change?
Better safe than sorry! (related: #19666)

screen shot 2018-05-17 at 11 07 22 am

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.js [hunk 1]
```diff
@@ -74,9 +74,20 @@ const show = (): void => {
     msg.show(makeHtml(template, bindableClassNames));
 };
 
+const test = {
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.js]: I'm guessing `test` is exposed to allow the tests to call `onAgree` and `binableClassNames`. There's a common pattern used throughout our js where we export a method named `_` to denote that it's for testing purposes.

eg. `export const _ = { onAgree,  binableClassNames };`.

File: static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.spec.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    jest.clearAllMocks();
+    setAdConsentState(0, null);
+    setAdConsentState(1, null);
+});
+beforeEach(() => {
+    Message.mockReset();
+    Message.prototype.show = jest.fn(() => true);
+    Message.prototype.hide = jest.fn(() => true);
+});
+afterEach(() => {
```
[the remark this answers is not in the document]
Comment (walaura) [reply static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.spec.js]: oops!

File: static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.js (hunk 1 above)
Comment (walaura) [reply static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.js]: ah alright!! i thought it was _test but the linter wouldn't let me name it like that so i went with test

File: static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.spec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    jest.clearAllMocks();
+    setAdConsentState(0, null);
+    setAdConsentState(1, null);
+});
+beforeEach(() => {
+    Message.mockReset();
+    Message.prototype.show = jest.fn(() => true);
+    Message.prototype.hide = jest.fn(() => true);
+    Message.prototype.show.mockRestore();
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.spec.js]: Previously `Message.prototype.show.mockRestore();` was in a duplicate `afterEach` did you mean to move it into the `beforeEach` when you removed the duplicate `afterEach` block?

Comment (walaura) [reply static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.spec.js]: 🙀 Yes! Nice catch. To be honest, this is all a copy paste job from other tests so let me put some order in this area, I think all of this belongs in `beforeEach()`

**Review by GHaberis — APPROVED** [review]
👍

**Pull request merged**